src/plugins/dashboard/public/application/legacy_app.js

```diff
@@ -127,14 +130,40 @@ export function initDashboardApp(app, deps) {
           $scope.create = () => {
             history.push(DashboardConstants.CREATE_NEW_DASHBOARD_URL);
           };
+          $scope.dashboardProviders = deps.dashboardProviders() || [];
+
           $scope.find = (search) => {
-            return service.find(search, $scope.listingLimit);
+            const searchRx = new RegExp(search, 'i');
+            const matchSearchToItem = (item) => {
+              const matchedTitle = !!item.title.match(searchRx);
+              const matchedDescription = !!item.description.match(searchRx);
+              const matchedType = !!item.type.match(searchRx);
+
+              return matchedTitle || matchedDescription || matchedType;
+            };
+
+            const $listItemFns = deps
+              .dashboardProviders()
+              .map((provider) => provider.listItemsFn());
+
+            const $listItems = concat($listItemFns).pipe(
+              mergeMap((items) => items),
+              filter(matchSearchToItem),
+              reduce(
+                (acc, item) => {
+                  return { hits: [...acc.hits, item], total: acc.total + 1 };
+                },
+                { hits: [], total: 0 }
+              )
+            );
+
+            return $listItems.toPromise();
           };
-          $scope.editItem = ({ id }) => {
-            history.push(`${createDashboardEditUrl(id)}?_a=(viewMode:edit)`);
+          $scope.editItem = ({ editUrl }) => {
+            window.location.href = editUrl;
```

Comment:
Browser-side routing does not work when the link routes outside of OSD-core into a plugin.

Reply:
I don't think so, almost all pages are plugins, including discover, dashboard, visualization etc. and they all do browser side routing.

Reply:
So far, in my testing, plugins that use HashRouter do respond correctly to a whole-url `history.push()` and only render their "default" page (whatever is configured within the plugin's HashRouter).

Only when `window.location` was used did the whole render, including HashRouter, behave correctly to the location change.

I would be happy to learn if there is some trick that I have missed?

Reply:
hmm, in case of jumping to another application doesn't work, can we keep the existing no page refresh behavior for dashboards, and only refresh the page when the destination URL is a different app?

Reply:
`window.location.href = ...` does not force a browser refresh.   So navigating from Dashboard-List to Dashboard-view will not reload page as both paths are inside Dashboard Plugin's  Hashrouter matching.

Navigations outside Dashboard's Hashrouter are then handled by browser, which then does GET on the new URL for server-side router to handle.   This is why navigating to Observability Panel results in a page-reload, because server-side routing is sending a new page on different matches to plugin routes.

@zengyan-amazon in our offline conversation, you indicated that based upon this result, the use of `window.location.href = ` is OK.